File: src/test/java/lrusso96/simplebiblio/StandardEbooksTest.java
package lrusso96.simplebiblio;

import lrusso96.simplebiblio.core.Ebook;
import lrusso96.simplebiblio.core.providers.standardebooks.StandardEbooks;
import lrusso96.simplebiblio.exceptions.BiblioException;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertNotEquals;
import static org.junit.Assert.assertNotNull;

public class StandardEbooksTest {

    @Test
    public void recentTest() throws BiblioException {
        StandardEbooks standard = new StandardEbooks();
        List<Ebook> ret = standard.getRecent();
        simpleTest(ret);
    }

    private void simpleTest(List<Ebook> ret) {
        assertNotEquals(0, ret.size());
        Ebook book = ret.get(0);
        assertNotNull(book.getTitle());
        assertNotNull(book.getAuthor());
        assertNotNull(book.getSummary());
        assertNotNull(book.getPublished());
        assertNotEquals(0, book.getDownload().size());
    }
}

File: src/test/java/lrusso96/simplebiblio/StandardEbooksTest.kt
package lrusso96.simplebiblio

import kotlinx.coroutines.runBlocking
import lrusso96.simplebiblio.core.Ebook
import lrusso96.simplebiblio.core.providers.standardebooks.StandardEbooks
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class StandardEbooksTest {

    @Test
    fun recentTest() = runBlocking { simpleTest(StandardEbooks().getRecent()) }

    private fun simpleTest(ret: List<Ebook>) {
        assertTrue(ret.isNotEmpty())
        val book = ret[0]
        assertNotNull(book.title)
        assertNotNull(book.author)
        assertNotNull(book.summary)
        assertNotNull(book.published)
        assertTrue(book.getDownloads().isNotEmpty())
    }
}